include/fmt/core.h

```diff
@@ -508,6 +508,7 @@ namespace internal {
 
 struct dummy_string_view { typedef void char_type; };
 dummy_string_view to_string_view(...);
+dummy_string_view to_string_view(const std::FILE *);
```

Comment:
What is this for?

Reply:
If I do `fmt::print(stderr, fmt::emphasis::bold, "");`, OR kicks in. It has a lot of candidates:

```c++
print(std::FILE *, const text_style &, format_str, args...); // this is what we want to get called
print(format_str, args...);
```
Normally, one would expect the first one to get called, but that's not the case. This is due to variadic templates being greedy. `fmt::emphasis::bold` is not a `text_style` and since `stderr` is a pointer to `T`, the second overload above is taken, which then fails with an assertion failure because `std::FILE` is not a valid character type.

I'm adding this so that fmt doesn't think that `stderr` is a valid format string - This never was an issue before because there were always exact matches.

Reply:
OK. Could you clarify this in a comment?

Reply:
@DanielaE Didn't want to change this for such a patch, but yeah that's the better solution. I'll update my PR then :) thanks

Reply:
In this case I'd rather split the PR into two: the first one laying the `cv T *` groundwork and then the second one actually addressing #944 one taking advantage of the first PR. This looks cleaner to me but at the end of the day it's up to Victor.

Reply:
Yeah, it looks like `std::char_traits` is defined for everything including `FILE` =), so we'll have to rely on some mechanism within {fmt} to detect char types. `to_string_view` is indeed our current way to detect string types and therefore character types, so unless there are better ideas let's go with the current `to_string_view`-based solution but please add a comment briefly explaining what's going on.

Reply:
😢 This sounds like a language or library defect. Characters are fundamental entities in the language and there are predicates to detect almost any kind of trait, but a well-defined predicate to detect the "character-ness" type trait seems to be missing. I've sifted through MSVC's library and found the catch-all primary template of `std::char_traits` which defines the  `char_type` member to `T`. Interestingly enough, only in this non-character case the `int_type` is defined to `long`. But this cannot be relied on because the specification in the standard is weasel-wordy in the case of non-required specializations.

Reply:
One more thing: `to_string_view(const std::FILE *)` should probably go to `fmt/color.h` since it only relevant for the `print` overloads there.